CHANGELOG.rst

```diff
@@ -0,0 +1,9 @@
+.. :changelog:
+
+History
+-------
+
+0.10.0 (unreleased)
+-------------------
+
+- Move to SpanContext, Reference, and `s/join/extract/g`
```

Comment:
Feel free to include or leave this file out as you want. I think in my original comment I was misusing the standard meaning of "changelog" and actually meant exactly what you have in the README (basically some "porting" tips for API incompatibility changes).  Thanks!